docs/UPGRADING.md

```diff
@@ -37,3 +37,33 @@ The `debug.xxx=yyy` definitions above may be supplied and overlayed in the usual
 - on `platform.txt`: definition here will be shared through all boards in the platform
 - on `boards.txt` as part of a board definition: they will override the global platform definitions
 - on `programmers.txt`: they will override the boards and global platform definitions if the programmer is selected
+
+### Binaries export must now be explicitly specified
+
+Previously, if the `--build-path` was not specified, compiling a Sketch would copy the generated binaries in
+`<sketch_folder>/build/<fqbn>/`, uploading to a board required that path to exist and contain the necessary binaries.
+
+The `--dry-run` flag is removed, `compile` now never copies generated binaries to the sketch folder. The
+`--export-binaries` (`-e`) flag is introduced to copy the binaries from the build folder to the sketch one. Specifying
+the `--output-dir` doesn't require using setting also the `--export-binaries` flag. A related settings and env var has
+been added to avoid the need to always specify the `--export-binaries` flag: `sketch.always_export_binaries` and
+`ARDUINO_SKETCH_ALWAYS_EXPORT_BINARIES`.
```

Comment:
```suggestion
The `--dry-run` flag was removed.

The default, `compile` does not copy generated binaries to the sketch folder. The `--export-binaries` (`-e`) flag was
introduced to copy the binaries from the build folder to the sketch one. `--export-binaries` is not required when using
the `--output-dir` flag. A related configuration keyh and environment variable has been added to avoid the need to
always specify the `--export-binaries` flag: `sketch.always_export_binaries` and
`ARDUINO_SKETCH_ALWAYS_EXPORT_BINARIES`.
```